core/include/crypto/crypto_secure_element.h

```diff
@@ -0,0 +1,44 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) Foundries Ltd. 2021 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+ /*
+  * This is the Cryptographic Secure Element API, part of the Cryptographic
+  * Provider API.
+  *
+  * These requests shall be handled in the secure element normally placed on
+  * a serial communication bus (SPI, I2C).
+  */
+#ifndef __CRYPTO_SECURE_ELEMENT_H
+#define __CRYPTO_SECURE_ELEMENT_H
+
+#include <tee_api_types.h>
+
+#if defined(CFG_NXP_SE05X)
+TEE_Result crypto_secure_element_enable_scp03(unsigned int rotate_keys);
+
+/*
+ * Type identifier for the APDU message as described by Smart Card Standard
+ * ISO7816-4 about ADPU message bodies decoding convention:
+ *
+ * https://cardwerk.com/smart-card-standard-iso7816-4-section-5-basic-organizations/#chap5_3_2
+ */
+enum crypto_apdu_type {
+	CRYPTO_APDU_CASE_1,
+	CRYPTO_APDU_CASE_2,
+	CRYPTO_APDU_CASE_2E,
+	CRYPTO_APDU_CASE_3,
+	CRYPTO_APDU_CASE_3E,
+	CRYPTO_APDU_CASE_4,
+	CRYPTO_APDU_CASE_4E,
+	CRYPTO_APDU_CASE_NO_HINT = 0xFF,
```

Comment:
no need for a specific value. I would put it in fisrt place in the enum.

Reply:
sure, sounds good